frontend/src/components/Button/Button.tsx

```diff
@@ -9,19 +10,89 @@ const styledButton = cva(styles.button, {
     stretch: {
       true: styles.stretch, // if stretch is true, add the stretch class to the button
     },
+    variant: {
+      primary: styles.primary,
+      secondary: styles.secondary,
+      white: styles.white,
+      outlined: styles.outlined,
+    },
+    size: {
+      medium: styles.sizeMedium,
+      large: styles.sizeLarge,
+    },
+    fullWidth: {
+      true: styles.fullWidth,
+    },
+    iconOnly: {
+      true: styles.iconOnly,
+    },
+  },
+  defaultVariants: {
+    variant: 'primary',
+    size: 'medium',
   },
 });
 
-const Button = ({ children, onClick, stretch = false, disabled = false }: ButtonProps) => {
+const Button = ({
+  children,
+  onClick,
+  stretch = false,
+  disabled = false,
+  variant = 'primary',
+  size = 'medium',
+  icon,
+  iconPosition = 'after',
+  iconOnly = false,
+  fullWidth = false,
+  type = 'button',
+  'aria-label': ariaLabel,
+  ...rest
+}: ButtonProps) => {
+  const renderedIcon =
+    typeof icon === 'string'
+      ? ICONS[icon] // URL string for SVG
+      : icon; // React node
+
   return (
     <button
       className={styledButton({
         stretch,
+        variant,
+        size,
+        fullWidth,
+        iconOnly,
       })}
       onClick={onClick}
       disabled={disabled}
+      type={type}
+      aria-label={ariaLabel}
+      {...rest}
     >
-      {children}
+      {iconOnly ? (
+        typeof renderedIcon === 'string' ? (
```

Comment:
Maybe we could simplify this part a bit, since nested ternary operators can be a bit hard to read and not a good practice. The code here feels more like configuration, so extracting a small helper function could make it cleaner and also reduce duplication, e.x.:
<img width="447" height="187" alt="Screenshot 2025-09-26 at 14 44 53" src="https://github.com/user-attachments/assets/a2354ebb-e0db-44cb-afdb-5908d2b8c013" />